Example-OSX/Example-OSX/Repository.swift

```diff
@@ -40,22 +38,43 @@ struct Repository {
     let forksCount: Int
     let url: URL
     let createdAt: Date
-
 }
 
-extension Repository: OperaDecodable, Decodable {
-
-    static func decode(_ json: Any) throws -> Repository {
-        return try Repository.init(  id: json => "id",
-                                     name: json => "name",
-                                     desc: json =>? "description",
-                                     company: json =>? ["owner", "login"],
-                                     language: json =>? "language",
-                                     openIssues: json => "open_issues_count",
-                                     stargazersCount: json => "stargazers_count",
-                                     forksCount: json => "forks_count",
-                                     url: URL(string: json => "url")!,
-                                     createdAt: json => "created_at")
+extension Repository: Decodable {
+        
+    private enum CodingKeys: String, CodingKey {
+        case id
+        case name
+        case desc = "description"
+        case language
+        case openIssues = "open_issues_count"
+        case stargazersCount = "stargazers_count"
+        case forksCount = "forks_count"
+        case url
+        case createdAt = "created_at"
+        case owner
+    }
+    
+    enum OwnerInfoKeys: String, CodingKey {
+        case company = "login"
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        id = try container.decode(Int.self, forKey: .id)
```

Comment:
You could use the `decode(.id)` here, right?

Reply:
ok